sdk/lusid/utilities/api_client_builder.py

```diff
@@ -19,26 +21,50 @@ def build(api_secrets_filename=None, okta_response_handler=None, api_configurati
         """
 
         # Load the configuration
-        configuration = api_configuration if api_configuration is not None else ApiConfigurationLoader().load(api_secrets_filename)
+        configuration = api_configuration if api_configuration is not None else ApiConfigurationLoader().load(
+            api_secrets_filename)
 
         encoded_password = pathname2url(configuration.password)
         encoded_client_id = pathname2url(configuration.client_id)
         encoded_client_secret = pathname2url(configuration.client_secret)
 
         # Prepare our authentication request
         token_request_body = f"grant_type=password&username={configuration.username}" \
-            f"&password={encoded_password}&scope=openid client groups offline_access" \
-            f"&client_id={encoded_client_id}&client_secret={encoded_client_secret}"
+                             f"&password={encoded_password}&scope=openid client groups offline_access" \
+                             f"&client_id={encoded_client_id}&client_secret={encoded_client_secret}"
         headers = {"Accept": "application/json", "Content-Type": "application/x-www-form-urlencoded"}
 
-        # Make our authentication request
-        okta_response = requests.post(configuration.token_url, data=token_request_body, headers=headers)
+        # extra request args
+        kwargs = {"headers": headers}
+
+        # use proxy if supplied
+        proxy_config = configuration.proxy_config
+        if proxy_config is not None:
+
+            proxy_url = proxy_config.address
+            if proxy_config.username is not None and proxy_config.password is not None:
+                index = proxy_config.address.index("://")
```

Comment:
Are you sure we should be massaging the values like this?

Reply:
this is substituting the username/password into a proxy url format (if they are supplied), open to suggestions

(note this has been refactored in the latest revision)